src/libs/actions/Welcome/OnboardingFlow.ts

```diff
@@ -44,11 +45,14 @@ function startOnboardingFlow(isPrivateDomain?: boolean) {
     if (focusedRoute?.name === currentRoute?.name) {
         return;
     }
-    navigationRef.resetRoot({
-        ...navigationRef.getRootState(),
-        ...adaptedState,
-        stale: true,
-    } as PartialState<NavigationState>);
+    // This should delay opening the onboarding modal so it does not interfere with the ongoing ReportScreen params changes
+    InteractionManager.runAfterInteractions(() => {
+        navigationRef.resetRoot({
+            ...navigationRef.getRootState(),
+            ...adaptedState,
+            stale: true,
+        } as PartialState<NavigationState>);
+    });
```

Comment:
Revert. This is not the solution that I meant.

You should wrap the [useOnboardingFlowRouter useEffect](https://github.com/Expensify/App/blob/927729c3ef12fb3a9a0878f5e712f2167d11f03a/src/hooks/useOnboardingFlow.ts#L34)

Reply:
Updated.